### PR TITLE
initial commit for basic  obstacle avoidance on the arudino

### PR DESCRIPTION
**Pull request by konradotto**
This pull request contains 3 files which should solve the task of basic obstacle avoidance with our smartcar.

*obstacleAvoidance.ino* contains a very basic Arduino program to prevent the car from colliding with an object frontally while driving at full speed. 

*usedpins.h*  declares our current pin-mapping. Since the pins will be used consistently it is best to store the mapping in one globally used header.

*constants.h* declares further global constants which don't regard the pinmapping. 

Please, review and let me know what you think!

### CODE REVIEW

**Review by jay-py — DISMISSED** [review]

File: arduino/obstacleAvoidance/usedpins.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// header-file defining the aruduino pins used 
+// for the different sensors and actuators
+
+#ifndef USEDPINS_H
+#define USEDPINS_H
+
+// left motor pins
+#define BRUSHED_LEFT_FORWARD_PIN	8
+#define BRUSHED_LEFT_BACKWARD_PIN	10
+#define BRUSHED_LEFT_ENABLE_PIN		9
```
Comment (platisd) [inline arduino/obstacleAvoidance/usedpins.h]: Why are you using macros to declare simple constants? Isn't `const int` or, if you want to go really low on memory, `const uint8_t` (aka `const unsigned char`) good enough for you? :cold_sweat:
Read more: [ES.31: Don't use macros for constants or "functions"](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-macros2).

File: arduino/obstacleAvoidance/obstacleAvoidance.ino [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// very basic obstacle avoidance
+// stop the car when an obstacle gets to close to the front sensor
+
+#include <Smartcar.h>
+#include "constants.h"
+
+// constants within this class (might need to be replaced with variables at some point)
+#define SPEED         100
+#define STOP_SPEED    0
+#define SAFETY_DIST   25      // not yet intended as the global standard - move to constants.h if it becomes that
```
Comment (platisd) [inline arduino/obstacleAvoidance/obstacleAvoidance.ino]: Why are you declaring constants as macros? See comment on `usedpins.h`.

**Review by platisd — CHANGES_REQUESTED** [review]
The sketch itself looks good. It would be great if you could change all those macros to proper constants. I would never forgive myself if you'd pick up bad habits under my watch! :anguished:

File: arduino/obstacleAvoidance/usedpins.h (hunk 1 above)
Comment (bartekspitza) [reply arduino/obstacleAvoidance/usedpins.h]: Changed all macros to constants in latest commit.

**Review by platisd — DISMISSED** [review]
:+1: 
Don't forget to also put the link to this pull request in canvas if you haven't done so already.

**Review by jay-py — APPROVED** [review]

**Pull request merged**